docs: update appinfo for webhook_listeners app

pReya opened the pull request:
## Summary
I found this app in my Nextcloud and it was very annoying that the App description was so short/non existent. The language around "webhooks" can be very ambiguous (e.g. there are incoming and outgoing webhooks, and the term "listener" can also be misjudged as "listening for incoming webhook").

So I was very confused by the short description of this App. Therefore I looked at the code and created a more detailed App description that might help some people understand easier what this app can be used for.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

pReya committed:
docs: update appinfo for webhook_listeners app

Signed-off-by: Moritz Stückler <moritz@bitbetter.de>

pReya committed:
fix: language

Signed-off-by: Moritz Stückler <moritz@bitbetter.de>

pReya committed:
fix: add link to docs

Signed-off-by: Moritz Stückler <moritz@bitbetter.de>

come-nc reviewed (commented):
Hey.

I get that you were annoyed but the name&shame on Mastodon is really not cool.

For context, this application was developped mainly as a dependency for https://github.com/nextcloud/flow

pReya commented:
Sorry @come-nc if my comment offended you. I've edited the toot and dialed it down. I did not want to name/shame individual developers or you. My criticism was supposed to go against Nextcloud in general and the priorities and processes within the project.

Thanks for giving the additional information about the reason why this app exists. I think this should also be part of the app description to explain, why a user might find this app in their Nextcloud (maybe because they're also using the Flow app).

joshtrichards committed:
chore(webhooks): Revise info.xml for Nextcloud Webhook Support

Updated app information for Nextcloud Webhook Support.

- Enhanced the description and summary to clarify functionality and usage.
- Add developer doc link

Signed-off-by: Josh <josh.t.richards@gmail.com>

joshtrichards commented:
- Revised the description and summary
- Added developer doc link
- Cleaned up the PR to prepare for merger

joshtrichards committed:
Merge branch 'master' into patch-1

Signed-off-by: Josh <josh.t.richards@gmail.com>

joshtrichards commented:
Replaced with #55412 to permit fixing conflicts/etc.
